spec/Appendix B -- Variable Batching.md

```diff
@@ -0,0 +1,170 @@
+
+### POST
+
+A **variable batching request** instructs the server to perform a query or mutation operation multiple times, once for each set of variables provided. The request MUST have a body that contains the values of the _variable batching request_ parameters encoded in the `application/graphql+jsonl` media type, or another media type supported by the server.
+
+A client MUST indicate the media type of a request body using the `Content-Type` header as specified in [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
+
+A server MUST support POST requests encoded with the `application/json` media type (as outlined in the GraphQL-over-HTTP specification).
+
+If the client does not supply a `Content-Type` header with a POST request, the server SHOULD reject the request using the appropriate `4xx` status code.
+
+### JSON Encoding
+
+When encoded in JSON, a _GraphQL-over-HTTP request_ is encoded as a JSON object
+(map), with the properties specified by the GraphQL-over-HTTP request:
+
+- {query} - the string representation of the Source Text of the Document as
+  specified in
+  [the Language section of the GraphQL specification](https://spec.graphql.org/draft/#sec-Language).
+- {operationName} - an optional string
+- **{variables}** - An array of JSON objects (maps), where each map corresponds to a set of variables to be used in the query.
+  names and the values of which are the variable values
+- {extensions} - an optional object (map)
+
+#### Example
+
+If we wanted to execute the following GraphQL query:
+
+```raw graphql example
+query ($id: ID!) {
+  user(id: $id) {
+    name
+  }
+}
+```
+
+With the following query variable sets:
+
+```json example
+[
+  {
+      "id": "QVBJcy5ndXJ1"
+  },
+  {
+      "id": "QVBJcy5ndXJ2"
+  },
+  {
+      "id": "QVBJcy5ndXJ3"
+  }
+]
+```
+
+This request could be sent via an HTTP POST to the relevant URL using the JSON
+encoding with the headers:
+
+```headers example
+Content-Type: application/json
+Accept: application/graphql-response+jsonl
+```
+
+And the body:
+
+```json example
+{
+  "query": "query ($id: ID!) {\n  user(id: $id) {\n    name\n  }\n}",
+  "variables": [
+    {
+      "id": "QVBJcy5ndXJ1"
+    },
+    {
+      "id": "QVBJcy5ndXJ2"
+    },
+    {
+      "id": "QVBJcy5ndXJ3"
+    }
+  ]
+}
+```
+
+### Response
+
+When a server receives a well-formed _variable batching request_, it MUST return a well‐formed stream of _GraphQL responses_. Each response in the stream corresponds to the result of validating and executing the requested operation with one set of variables. The server's response stream describes the outcome of each operation, including any errors encountered during the execution of the request.
```

Comment:
I don't think we should use the word "stream" here. I think "list" is fine; I think the default media type would likely be e.g. `application/graphql-response-list+json` (a JSON array of GraphQL responses) but `application/jsonl` would allow for easier early parsing of responses when the client is stream-capable.